pkg/ext-proc/backend/pod_reconciler.go

```diff
@@ -0,0 +1,73 @@
+package backend
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"inference.networking.x-k8s.io/gateway-api-inference-extension/api/v1alpha1"
+	logutil "inference.networking.x-k8s.io/gateway-api-inference-extension/pkg/ext-proc/util/logging"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type PodReconciler struct {
+	client.Client
+	Datastore *K8sDatastore
+	Scheme    *runtime.Scheme
+	Record    record.EventRecorder
+}
+
+func (c *PodReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	inferencePool, err := c.Datastore.getInferencePool()
+	if err != nil {
+		klog.V(logutil.DEFAULT).Infof("Skipping reconciling EndpointSlice because the InferencePool is not available yet: %v", err)
+		return ctrl.Result{Requeue: true, RequeueAfter: time.Second}, nil
```

Comment:
No need to requeue since we will have to list on llmserverpool changes to address selector updates.

Reply:
There is no need to requeue, it will re-queue all the pods in the cluster. The problem this tries to solve is already addressed by the list we do on inferencePool events.